lib/block_view.go

```diff
@@ -7349,53 +7731,62 @@ func (bav *UtxoView) _connectAcceptNFTBid(
 	nftPostEntry.NumNFTCopiesForSale--
 	bav._setPostEntryMappings(nftPostEntry)
 
-	// Add an operation to the list at the end indicating we've connected an NFT bid.
-	utxoOpsForTxn = append(utxoOpsForTxn, &UtxoOperation{
-		Type:                      OperationTypeAcceptNFTBid,
+	// Create an Operation to add to the end of the list. Fill all fields except the type which depends upon
+	// if this is an AcceptNFTBid transaction or an NFTBid transaction.
+	transactionUtxoOp := &UtxoOperation{
 		PrevNFTEntry:              prevNFTEntry,
 		PrevPostEntry:             prevPostEntry,
 		PrevCoinEntry:             &prevCoinEntry,
 		DeletedNFTBidEntries:      deletedBidEntries,
 		NFTPaymentUtxoKeys:        nftPaymentUtxoKeys,
 		NFTSpentUtxoEntries:       spentUtxoEntries,
 		PrevAcceptedNFTBidEntries: prevAcceptedBidHistory,
-
 		// Rosetta fields.
 		AcceptNFTBidCreatorPublicKey: nftPostEntry.PosterPublicKey,
 		AcceptNFTBidBidderPublicKey: bidderPublicKey,
 		AcceptNFTBidCreatorRoyaltyNanos: creatorCoinRoyaltyNanos,
-	})
+	}
+	if txMeta.IsAcceptNFTBid {
```

Comment:
You could actually put `Type: args.Type` if you put the OperationType in the helper struct instead of the bool.

Reply:
`OperationType` and `TxnType` are two different types unfortunately.